pandas/core/frame.py

```diff
@@ -9241,7 +9241,12 @@ def mode(
         def f(s):
             return s.mode(dropna=dropna)
 
-        return data.apply(f, axis=axis)
+        data = data.apply(f, axis=axis)
+        # Ensure index is type stable (should always use int index)
+        if data.empty:
+            data.index = Index([], dtype="int64")
```

Comment:
is this right for 32-bit machines?

Reply:
Changed to `int`, thanks for catching that